packages/web/queries/complex/pools.ts

```diff
@@ -0,0 +1,325 @@
+
+        const cosmwasmPools = await getCosmwasmPools();
+
+        // prepend cosmwasm pools
+        return (cosmwasmPools as PoolRaw[]).concat(
+          queryPoolRawResults.filter(
+            (
+              poolRaw
+            ): poolRaw is
+              | StablePoolRaw
+              | ConcentratedLiquidityPoolRaw
+              | WeightedPoolRaw => !!poolRaw
+          )
+        );
+      } catch (e) {
+        console.error(e);
+
+        // fall back to pools query on node
+        return await getPoolsFromNode();
+      }
+    },
+    ttl: 30 * 1000, // 30 seconds
+  });
+}
+
+export async function queryPoolRawFromFilteredPool(
+  filteredPool: FilteredPoolsResponse["pools"][0]
+): Promise<
+  StablePoolRaw | ConcentratedLiquidityPoolRaw | WeightedPoolRaw | undefined
+> {
+  // deny pools containing tokens with gamm denoms
+  if (
+    Array.isArray(filteredPool.pool_tokens) &&
+    filteredPool.pool_tokens.some(
+      (token) => "denom" in token && token.denom.includes("gamm")
+    )
+  ) {
+    return;
+  }
+
+  /** Metrics common to all pools. */
+  const poolMetrics: {
+    liquidityUsd: number;
+    liquidity24hUsdChange: number;
+
+    volume24hUsd: number;
+    volume24hUsdChange: number;
+
+    volume7dUsd: number;
+  } = {
+    liquidityUsd: filteredPool.liquidity,
+    liquidity24hUsdChange: filteredPool.liquidity_24h_change,
+    volume24hUsd: filteredPool.volume_24h,
+    volume24hUsdChange: filteredPool.volume_24h_change,
+    volume7dUsd: filteredPool.volume_7d,
+  };
+
+  if (
+    filteredPool.type === "osmosis.concentratedliquidity.v1beta1.Pool" &&
+    !Array.isArray(filteredPool.pool_tokens)
+  ) {
+    if (!filteredPool.pool_tokens.asset0 || !filteredPool.pool_tokens.asset1)
+      return;
+
+    const token0 = filteredPool.pool_tokens.asset0.denom;
+    const token1 = filteredPool.pool_tokens.asset1.denom;
+
+    return {
+      "@type": `/${filteredPool.type}`,
+      address: filteredPool.address,
+      id: filteredPool.pool_id.toString(),
+      current_tick_liquidity: filteredPool.current_tick_liquidity,
+      token0,
+      token0Amount: makeCoinFromToken(filteredPool.pool_tokens.asset0).amount,
+      token1,
+      token1Amount: makeCoinFromToken(filteredPool.pool_tokens.asset1).amount,
+      current_sqrt_price: filteredPool.current_sqrt_price,
+      current_tick: filteredPool.current_tick,
+      tick_spacing: filteredPool.tick_spacing,
+      exponent_at_price_one: filteredPool.exponent_at_price_one,
+      spread_factor: filteredPool.spread_factor,
+      ...poolMetrics,
+    };
+  }
+
+  const sharePoolBase = {
+    "@type": `/${filteredPool.type}`,
+    id: filteredPool.pool_id.toString(),
+    pool_params: {
+      exit_fee: new Dec(filteredPool.exit_fees.toString())
+        .mul(DecUtils.getTenExponentN(-2))
+        .toString(),
+      swap_fee: new Dec(filteredPool.swap_fees.toString())
+        .mul(DecUtils.getTenExponentN(-2))
+        .toString(),
```

Comment:
Do we need to use @keplr/units for these functions? Btw, how much does Keplr bloat the bundle size of the functions?

Reply:
I can remove those. But how high prio are concerns for bigger bundles on server? I imagine it's low impact since it's not going over wire